network/src/utils/mod.rs

```diff
@@ -15,6 +15,29 @@ pub(crate) fn run_command(cmd: &str, args: &[&str]) -> Result<String, KudoNetwor
     wrap_command_output(output, cmd, args)
 }
 
+pub(crate) fn get_default_interface_name() -> Result<String, KudoNetworkError> {
+    match default_net::get_default_interface() {
+        Ok(default_interface) => Ok(default_interface.name),
+        Err(e) => Err(KudoNetworkError::DefaultNetworkInterfaceError(format!(
+            "Could not find default interface : {}",
+            e
+        ))),
+    }
+}
```

Comment:
You should use a `map().map_err()`combination for this.

Reply:
You have not addressed that one yet?

Reply:
My bad, it's corrected now